### PR TITLE
Search: Use react-autosuggest with Fuse.js for field search

### PR DESCRIPTION
**Pull request by wei2912**
The search bar now displays suggestions in a nicely styled manner (up to 10 suggestions, split across the sections) and the filtering works decently.

Todo:
- [x] Refine the splitting of suggestions logic so that if one section got < max amount, ramp up the max
- [x] Style the search bar properly
- [x] Add proper filtering logic for field search

### CODE REVIEW

**Comment by wei2912** [comment]
@joyceerhl Thanks for the review! I'll look through your comments tomorrow.

For the Lunr.js filtering logic, we should go with https://github.com/olivernn/lunr.js/issues/366 and use the direct query solution.

I'd like to propose that we have two types of searches:
1. Suggestion search - this uses field-specific search with Lunr.js and leads to specific results.
2. Free search - if the user doesn't select anything, then Lunr.js will search based on whatever the user types.
My original plan was to combine both, but I think that would be too troublesome to implement and won't lead to much improvement. We can get the first one settled first (if Lunr.js causes trouble then I guess a good old `.contains` won't hurt), and the second one up another time.

**Comment by wei2912** [comment]
Please feel free to add commits directly to this branch, as I'll no longer be rebasing and force pushing.

**Comment by wei2912** [comment]
I have left a bunch of comments that explain how to write a solution to this *without requiring the use of `Lunr.js`* (cos that thing is a bit hard to customize for this setup due to tokenisation). Would be great if we could push out that solution with the proper styling in place.

**Comment by wei2912** [comment]
Latest commit implements the filter functionality for fields correctly, just got to style.

**Comment by wei2912** [comment]
@joyceerhl Thanks for the changes! For point 3 of your comments,

> Can we left-align (instead of center-align) the options in the dropdown list?

Not sure if `left-align` would look better:

image

What do you think?

File: src/search.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { mentors } from "./mentors.json";
+
+const fieldSearch = ({ field, value }) =>
+  mentors
+    .map((mentor, index) => index)
+    .filter((index) => mentors[index][field] === value);
```
Comment (bnjmnt4n) [inline src/search.js:6]: This can probably be simplified to `.map((mentor) => mentor[field] === value)`.

Comment (wei2912) [reply src/search.js:6]: The intention was to get a list of indexes at the end without changing the `index` :P have to say this index business with the `mentors` is pretty annoying and I should get to generating an unique ID for each

**Comment by wei2912** [comment]
I've narrowed the suggestions and left-aligned the title. Hopefully shld work better on mobile now

File: src/search.js (hunk 1 above)
Comment (bnjmnt4n) [reply src/search.js:6]: Oh right, scratch that :)

File: src/components/search-bar.css [hunk 2]
```diff
@@ -55,8 +55,7 @@
 
 .react-autosuggest__suggestion {
   cursor: pointer;
-  padding: 10px 20px;
-  text-align: left;
+  padding: 5px 20px;
```
Comment (bnjmnt4n) [inline src/components/search-bar.css:58]: Now it looks a little too small - maybe something like 8px?

Comment (wei2912) [reply src/components/search-bar.css:58]: Oof, will expand it...

**Pull request merged**